microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,221 @@
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
+            self.log.info("Packages installed for Ubunutu")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "mv /usr/bin/meson /usr/bin/meson.bak")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+            self._execute_expect_zero(node, "pip3 install --upgrade ninja")
+
+        elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self._execute_expect_zero(
+                node, "yum update -y --disablerepo='*' --enablerepo='*microsoft*'"
+            )
+            node.os.install_packages(
+                ["groupinstall", "'Infiniband Support'"], signed=False
+            )  # todo gross hack to support groupinstall
+            for package in self._redhat_packages:
+                node.os.install_packages(package)
+            result = node.execute(
+                "dracut --add-drivers 'mlx4_en mlx4_ib mlx5_ib' -f"
+            )  # add mellanox drivers
+            self.log.debug("\n".join([result.stdout, result.stderr]))
+            self._execute_expect_zero(node, "systemctl enable rdma")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+
+            self._execute_expect_zero(
+                node,
+                f"wget {self._ninja_url}",
+            )
+            self._execute_expect_zero(
+                node, "unzip ninja-linux.zip && mv ninja /usr/bin/ninja"
+            )
+            self._execute_expect_zero(node, "pip3 install --upgrade pyelftools")
+
+    def _execute_expect_zero_with_path(
+        self, node: Node, cmd: str, path: PurePath
+    ) -> str:
+        result = node.execute(cmd, sudo=True, cwd=path, shell=True)
+        result.assert_exit_code()
+        self.log.info(result.stdout)  # TODO: debug
+
+        return result.stdout
+
+    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
+        return self._execute_expect_zero_with_path(node, cmd, node.working_path)
+
+    def _hugepages_init(self, node: Node) -> None:
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge")
+        self._execute_expect_zero(node, "mkdir -p /mnt/huge-1G")
+        self._execute_expect_zero(node, "mount -t hugetlbfs nodev /mnt/huge")
+        self._execute_expect_zero(
+            node, "mount -t hugetlbfs nodev /mnt/huge-1G -o 'pagesize=1G'"
+        )
+
+    def _hugepages_enable(self, node: Node) -> None:
+        self._execute_expect_zero(
+            node,
+            "echo 4096 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
+        )
+        self._execute_expect_zero(
+            node,
+            "echo 1 > /sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
```

Comment:
OP is right about having a specific runbook requirement here, or we'd have to have some automatic ARM templates to kick in per test suite/criteria. These are huge mem requirements, not met by your average Azure VM.

Reply:
That's a good point, DPDK in general has some requirements like needing >=8 cores so I'll have to add something for memory too.